test/autoCompletion.test.ts

```diff
@@ -34,7 +34,19 @@ describe('Auto Completion Tests', () => {
     yamlSettings = settings;
   });
 
-  function parseSetup(content: string, position: number): Promise<CompletionList> {
+  /**
+   * Generates a completion list for the given document and caret (cursor) position.
+   * @param content The content of the document.
+   * @param position The position of the caret in the document.
+   * Alternatively, can be omitted if the caret is located in the content using the symbol `|:|`.
+   * @returns A list of valid completions.
+   */
+  function parseSetup(content: string, position?: number): Promise<CompletionList> {
+    if (typeof position === 'undefined') {
+      position = content.indexOf('|:|');
+      content = content.replace('|:|', '');
+    }
```

Comment:
This is the critical change
If the `position` is not specified, we look inside the `content` to derive the `position` using the index of the `|:|` token